### PR TITLE
Move all node location and path conversion logic to NavNode/SourceFile

### LINKED ISSUE #9: Search results show URL encoded location
image

### LINKED ISSUE #13: `.order` will hide files with hyphens
I think there may be a bug here where, if your `.order` contains `Release-Notes` and `Release-Notes` does exist, it will still filter that file because it doesn't match it correctly due to escaping.

### PR DESCRIPTION
**Pull request by ricardoboss**

### CODE REVIEW

**Comment by ricardoboss** [comment]
Maybe the code for en- and decoding the location/node properties should be moved somewhere else. Maybe introduce some kind of `INavNodeProcessingService` or extend the `INodeFlaggingService`?

**Comment by chucker** [comment]
Can't those methods simply be on the node itself?

**Comment by ricardoboss** [comment]
I'm new to this whole OOP thing, ok? (yeah, you're right)

File: TRENZ.Docs.API/Models/NavNode.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    /// <summary>
+    /// Whether or not this node has a corresponding content file.
+    /// </summary>
+    public bool HasContent { get; set; }
+
+    /// <summary>
+    /// The display name of the node in the tree. This is the last part of <see cref="Location" />.
+    /// </summary>
+    public string NodeName => LocationParts.Last();
```
Comment (chucker) [inline TRENZ.Docs.API/Models/NavNode.cs:54]: thanks for documenting these

**Review by chucker — APPROVED** [review]

**Pull request merged**